utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -55,194 +82,356 @@ class BioReader():
+            assert np.any(np.less_equal(0, ct)),\
+                'At least one of the {}-indices was less than 0.'.format(axis)
+            assert len(ct) != 0,\
+                'At least one {}-index must be selected.'.format(axis)
         return ct
-    
-    def read_image(self,X=None,Y=None,Z=None,C=None,T=None,series=None):
-        '''
-        image = BioReader.open()
-        
-        -Inputs-
-        X,Y,Z -     If None, then load all values in the given dimension. If a
-                    2-tuple, load values within the range.
-        C,T -       If None, then load all channels/timepoints, respectively. If a
-                    list or tuple, load indicated channels/timepoints.
-        series -    Placeholder, in the future will open multi-series images
-        
-        -Outputs-
-        image - a 5D numpy array with dimensions corresponding to (y,x,z,channel,timepoint)
-        
-        -Example-
-        # Import javabridge and start the vm
-        import javabridge
-        javabridge.start_vm(class_path=bioformats.JARS)
-        
-        # Path to bioformats supported image
-        image_path = 'path/to/image'
-        
-        # Create the BioReader object
-        bf = BioReader(image_path)
-        
-        # Load the full image
-        image = bf.read_image()
-        
-        # Only load the first 256x256 pixels, will still load all Z,C,T dimensions
-        image = bf.read_image(X=(0,256),Y=(0,256))
-        
-        # Only load the second channel
-        image = bf.read_image(C=[1])
-        
-        # Done executing program, so kill the vm. If the program needs to be run
-        # again, a new interpreter will need to be spawned to start the vm.
-        javabridge.kill_vm()
-        '''
+
+    def read_image(self, X=None, Y=None, Z=None, C=None, T=None, series=None):
+        """read_image Read the image
+
+        Read the image. A 5-dimmensional numpy.ndarray is always returned.
+
+        Args:
+            X ([tuple,list], optional): 2-tuple indicating the (min,max) range of
+                pixels to load. If None, loads the full range.
+                Defaults to None.
+            Y ([tuple,list], optional): 2-tuple indicating the (min,max) range of
+                pixels to load. If None, loads the full range.
+                Defaults to None.
+            Z ([tuple,list], optional): 2-tuple indicating the (min,max) range of
+                pixels to load. If None, loads the full range.
+                Defaults to None.
+            C ([tuple,list], optional): tuple or list of values indicating channel
+                indices to load. If None, loads the full range.
+                Defaults to None.
+            T ([tuple,list], optional): tuple or list of values indicating timepoints
+                to load. If None, loads the full range.
+                Defaults to None.
+            series (tuple, optional): Placeholder. Currently does nothing.
```

Comment:
defined here but not above.  Still should probably move definition to above comment.

Reply:
`series` is the input argument for this method. If `series` were a method, it would belong above.